app/src/pages/explorer/curate/spreadsheet/spreadsheet-upload-script.js

```diff
@@ -0,0 +1,195 @@
+import DropZone from '@/components/curate/FileDrop.vue'
+import FilePreview from '@/components/curate/FilePreview.vue'
+import LoginRequired from '@/components/LoginRequired.vue'
+import Dialog from '@/components/Dialog.vue'
+import CurateNavBar from '@/components/curate/CurateNavBar.vue'
+import Spinner from '@/components/Spinner.vue'
+import useFileList from '@/modules/file-list'
+import { VERIFY_AUTH_QUERY, USER_DATASET_IDS_QUERY, CREATE_DATASET_MUTATION } from '@/modules/gql/dataset-gql'
+import { mapGetters, mapMutations } from 'vuex'
+
+// Create separate file objects for spreadsheet vs supplementary files
+const spreadsheetFn = useFileList()
+const suppFn = useFileList()
+
+export default {
+  name: 'SpreadsheetHome',
+  components: {
+    dialogbox: Dialog,
+    DropZone,
+    FilePreview,
+    CurateNavBar,
+    Spinner,
+    LoginReq: LoginRequired
+  },
+  data () {
+    return {
+      auth: true,
+      verifyUser: null,
+      loading: false,
+      uploadInProgress: true,
+      selectedDataset: {
+        label: '',
+        id: null
+      },
+      spreadsheetFiles: spreadsheetFn.files,
+      suppFiles: suppFn.files,
+      uploadedFiles: null,
+      title: null,
+      doi: null,
+      active: 'first',
+      first: false,
+      second: false,
+      third: false,
+      fourth: false,
+      fifth: false,
+      sixth: false,
+      isInvalidForm: false,
+      dialog: {
+        title: '',
+        type: null,
+        size: 60,
+        disableClose: false
+      },
+      navRoutes: [
+        {
+          label: 'Curate',
+          path: '/explorer/curate'
+        },
+        {
+          label: 'Spreadsheet',
+          path: '/explorer/curate/spreadsheet'
+        }
+      ]
+    }
+  },
+  props: {
+    datasetId: {
+      type: String
+    }
+  },
+  computed: {
+    ...mapGetters({
+      userId: 'auth/userId',
+      isAuthenticated: 'auth/isAuthenticated',
+      dialogBoxActive: 'dialogBox'
+    })
+  },
+  methods: {
+    addSpreadsheet: spreadsheetFn.addFiles,
+    removeSpreadsheet: spreadsheetFn.removeFile,
+    modStatSpreadsheet: spreadsheetFn.modifyStatus,
+    addSupp: suppFn.addFiles,
+    removeSupp: suppFn.removeFile,
+    modStatSupp: suppFn.modifyStatus,
+    ...mapMutations({
+      toggleDialogBox: 'setDialogBox'
+    }),
+    async navBack () {
+      this.$router.back()
+    },
+    async createDataset () {
+      await this.$apollo.mutate({
+        mutation: CREATE_DATASET_MUTATION,
+        variables: {
+          input: {
+            datasetId: `${this.datasetId}`,
+            files: this.uploadedFiles.files,
+            title: this.title,
+            doi: this.doi
+          }
+        }
+      }).then((data) => {
+        return data
+      }).catch((error) => {
+        console.error('error:', error)
+      })
```

Comment:
NIT: Can we refactor this, and pass the await into a const with a try & catch block?

Reply:
I realized I don't actually need the await here. For the refactor, I'm copying the [Vue Apollo example](https://apollo.vuejs.org/guide/apollo/mutations.html)